### PR TITLE
feat: Utilities for constructing BIDSFiles and FileTrees

### PR DESCRIPTION
**Pull request by effigies**
We were starting to excessively reimplement creation of BIDSFiles, and creating FileTrees is cumbersome. I had started to abstract out the FileTree generation from the browser implementation last week, so I finished that.

`src/files/filetree.ts` exports:

* `pathToFile(path: string): BIDSFile`
* `filesToTree(fileList: BIDSFile[]): FileTree`

And a convenience function that simply combines the two in the obvious way:

```typescript
export function pathsToTree(paths: string[]): FileTree {
  return filesToTree(paths.map(pathToFile))
}
```

### CODE REVIEW

File: bids-validator/src/files/browser.test.ts [hunk 1]
```diff
@@ -28,7 +28,7 @@ Deno.test('Browser implementation of FileTree', async (t) => {
       new TestFile(['flat test dataset'], 'README.md', 'ds/README.md'),
     ]
     const tree = await fileListToTree(files)
```
Comment (effigies) [inline bids-validator/src/files/browser.test.ts]:
```suggestion
    const tree = fileListToTree(files)
```

File: bids-validator/src/files/browser.test.ts [hunk 2]
```diff
@@ -61,9 +61,9 @@ Deno.test('Browser implementation of FileTree', async (t) => {
       ),
     ]
     const tree = await fileListToTree(files)
```
Comment (effigies) [inline bids-validator/src/files/browser.test.ts]:
```suggestion
    const tree = fileListToTree(files)
```

File: bids-validator/src/files/browser.test.ts [hunk 3]
```diff
@@ -90,7 +90,7 @@ Deno.test('Spread copies of BIDSFileBrowser contain name and path properties', a
     new TestFile(['flat test dataset'], 'README.md', 'ds/README.md'),
   ]
   const tree = await fileListToTree(files)
```
Comment (effigies) [inline bids-validator/src/files/browser.test.ts]:
```suggestion
  const tree = fileListToTree(files)
```

**Review by effigies — COMMENTED** [review]
Missed these.

File: web/src/App.tsx [hunk 4]
```diff
@@ -42,7 +42,7 @@ function App() {
     const dirHandle = await directoryOpen({
       recursive: true,
     })
-    const fileTree = await fileListToTree(dirHandle)
```
Comment (rwblair) [inline web/src/App.tsx:45]: I had to add this await back in, fileListToTree was returning a promise. Could be a me problem though.

File: bids-validator/src/files/browser.ts [hunk 5]
```diff
@@ -48,40 +49,8 @@ export class BIDSFileBrowser implements BIDSFile {
 /**
  * Convert from FileList (created with webkitDirectory: true) to FileTree for validator use
  */
-export function fileListToTree(files: File[]): Promise<FileTree> {
+export function fileListToTree(files: File[]): FileTree {
```
Comment (effigies) [inline bids-validator/src/files/browser.ts:52]: It seems like it shouldn't be?

Comment (rwblair) [reply bids-validator/src/files/browser.ts:52]: Ah yeah didn't notice it getting sourced from web/dist/main.ts My build.ts isn't working and dist must be outdated.

Comment (effigies) [reply bids-validator/src/files/browser.ts:52]: Yeah, you'll need #2088 before this will work.

**Comment by effigies** [comment]
Rebased on #2088 for easier testing.

**Pull request merged**